### PR TITLE
reef: json integration: identifying new data-source and execute json flow.

### PR DESCRIPTION
**Pull request by cbodley**
reef backport for https://github.com/ceph/ceph/pull/49411

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/radosgw/s3select.rst [hunk 1]
```diff
@@ -496,6 +503,13 @@ Output Serialization
    | **FieldDelimiter** -> (string)
    | The value used to separate individual fields in a record. You can specify an arbitrary delimiter.
 
+scan range option
```
Comment (anthonyeleven) [inline doc/radosgw/s3select.rst:506]: Caps?

File: doc/radosgw/s3select.rst [hunk 2]
```diff
@@ -496,6 +503,13 @@ Output Serialization
...
+~~~~~~~~~~~~~~~~~
+
+   | The scan range option is a part of AWS-CLI syntax, it enables to scan and process only the selected part of the object. 
+   | This option reduces the amount of IO operations (by skipping).
```
Comment (anthonyeleven) [inline doc/radosgw/s3select.rst:510]: s/amount/number/

File: doc/radosgw/s3select.rst [hunk 3]
```diff
@@ -530,6 +544,94 @@ CSV parsing behavior
...
 
+JSON
+--------------------
+
+         | a JSON reader has been integrated with the s3select-engine, which allows the client to use SQL statements to scan and extract information from JSON documents. 
+         | It should be noted that the data readers and parsers for CSV, Parquet, and JSON documents are separated from the SQL engine itself, so all of these readers use the same SQL engine.
+
+         | It's important to note that values in a JSON document can be nested in various ways, such as within objects or arrays.
+         | These objects and arrays can be nested within each other without any limitations.
+         | upon using SQL to query a specific value in a JSON document, the user needs to use a specific syntax to describe the location of the value.
```
Comment (anthonyeleven) [inline doc/radosgw/s3select.rst:555]: s/upon/When/

File: doc/radosgw/s3select.rst [hunk 4]
```diff
@@ -530,6 +544,94 @@ CSV parsing behavior
...
+         | This is because the standard "select column from object" syntax will not work.
```
Comment (anthonyeleven) [inline doc/radosgw/s3select.rst:556]: SELECT COLUMN?

File: doc/radosgw/s3select.rst [hunk 5]
```diff
@@ -530,6 +544,94 @@ CSV parsing behavior
...
+         | Instead, the user must use a path in the SELECT statement to tell the JSON reader where the value is located.
+
+         | The SQL engine processes the SELECT statement in a row-based fashion.
+         | It uses the columns specified in the statement to perform its projection calculation, and each row contains values for these columns.
+         | In other words, the SQL engine processes each row one at a time(and aggregates results), using the values in the columns to perform its SQL calculations.
```
Comment (anthonyeleven) [inline doc/radosgw/s3select.rst:561]: Space after `time`

**Comment by cbodley** [comment]
@anthonyeleven i always appreciate your reviews, but backport PRs are cherry-picked directly from main. if we need to make these changes, they'd need to happen on the main branch first

**Comment by cbodley** [comment]
^ cc @galsalomon66

**Comment by anthonyeleven** [comment]
Ack.  I hadn't seen the `main` PR here so I missed that this was a backport.

**Comment by cbodley** [comment]
> Ack. I hadn't seen the `main` PR here so I missed that this was a backport.

would you be willing to open a PR against main with your requested changes? those could merge without QA and i'd gladly include them here

**Comment by anthonyeleven** [comment]
sure

File: doc/radosgw/s3select.rst (hunk 4 above)
Comment (galsalomon66) [reply doc/radosgw/s3select.rst:556]: ok,
this paragraph will be rephrased.

**Comment by anthonyeleven** [comment]
I created https://github.com/ceph/ceph/pull/50384 and could not resist the urge to touch all manner of other subtle things.
Suggest merging this and we'll deal with the the above independently.

**Review by anthonyeleven — APPROVED** [review]

**Comment by cbodley** [comment]
passed qa in https://pulpito.ceph.com/cbodley-2023-03-06_17:47:36-rgw-wip-cbodley-testing-distro-default-smithi/

**Comment by cbodley** [comment]
jenkins test api

**Comment by cbodley** [comment]
hello? jenkins?

**Review by alimaredia — APPROVED** [review]

**Pull request merged**